src/Client/ClientBase.cpp

```diff
@@ -67,13 +67,56 @@ namespace ErrorCodes
     extern const int NO_DATA_TO_INSERT;
     extern const int UNEXPECTED_PACKET_FROM_SERVER;
     extern const int INVALID_USAGE_OF_INPUT;
+    extern const int CANNOT_SET_SIGNAL_HANDLER;
 }
 
 }
 
 namespace DB
 {
 
+
+std::atomic_flag exit_on_signal = ATOMIC_FLAG_INIT;
```

Comment:
It initializes flag to false, but it should be true when program starts.

Reply:
I know, it will be initialized to true in setupSignalHandler.